app/src/main/java/com/keylesspalace/tusky/util/FontScaleContextWrapper.kt

```diff
@@ -0,0 +1,43 @@
+package com.keylesspalace.tusky.util
+
+import android.content.Context
+import android.content.ContextWrapper
+
+class FontScaleContextWrapper(base: Context?) : ContextWrapper(base) {
+    companion object {
+        const val TAG = "FontScaleContextWrapper"
+
+        /**
+         * Wraps `context`, setting the correct `fontScale` from `fontScaleRatio`.
+         */
+        fun wrap(context: Context, fontScaleRatio: Float): FontScaleContextWrapper {
+            val configuration = context.resources.configuration
+
+            // Can't simply adjust the `fontScale` in `context`, because that will contain the
+            // result of previous adjustments. E.g., going from 100% to 80% to 100% does not return
+            // you to the original 100%, it leaves it at 80%.
+            //
+            // The application context is unaffected by changes to the base context, so always use
+            // the `fontScale` from that as the initial value to multiply. This will also contain
+            // any changes to the font scale from "Settings > Display > Font size" in the device
+            // settings.
+            val appConfiguration = context.applicationContext.resources.configuration
```

Comment:
This line does not use "fontScale"?

(How does this comment relate to the code?)

Reply:
It's explaining the use of `appConfiguration.fontScale` a few lines later.

This could also be:

```
val appFontScale = context.applicationContext.resources.configuration.fontScale
[...]
configuration.fontScale = appFontScale * fontScaleRatio
```

It's a simple change, if you think it makes the code clearer.

Reply:
I would probably combine the two comments and shorten them to something like

// Always adjust based on the initial value of fontScale; this contains the device settings and also avoids problems with an unaffected application context.
//
// NOTE one could also try to change the densityDpi (see commented code below). However this has further problems (e. g. dialogs being clipped) for larger size.